server/controllers/payment.js

```diff
@@ -0,0 +1,139 @@
+const asyncHandler = require("express-async-handler");
+const Payment = require("../models/Payment");
+const stripe = require("stripe")(process.env.STRIPE_SECRET_KEY);
+
+// @route GET /payments
+// @desc Get all payment from user
+// @access Private
+exports.getAllPayments = asyncHandler(async (req, res, next) => {
+  const payment = await Payment.find({ userId: req.user.id });
+  if (payment) {
```

Comment:
There's no need for this check. The mongoose `find()` method returns an array. So you just return it to the client regardless of what the content is.